### PR TITLE
sw-6820 amend routing for pagination tests to work

### PR DESCRIPTION
**Pull request by kate-49**

### CODE REVIEW

File: cypress/e2e/client-tasks/pagination.cy.js [hunk 1]
```diff
@@ -9,19 +9,28 @@ describe("Pagination", () => {
     beforeEach(() => {
       cy.get("#top-nav .display-rows").select('25')
           .invoke('val').should('contain', 'per-page=25')
+      cy.get(".moj-team-banner__container > .govuk-form-group > .govuk-select").select("Lay Team 2 - (Supervision)")
     })
 
     it("will not show previous arrow on page 1 but will show it on other pages", () => {
```
Comment (devkokov) [inline cypress/e2e/client-tasks/pagination.cy.js:15]: I could be wrong but I don't think the test is doing what it says it does

File: json-server/config/routes.json [hunk 2]
```diff
@@ -4,14 +4,21 @@
...
-    "/users/:id/tasks/:id": "/assign-tasks-to-casemanager"
+    "/users/:id/tasks/:id": "/assign-tasks-to-casemanager",
+
+    "/assignees/teams/tasks\\?teamIds[]=22&filter=status\\:Not+started&limit=25&page=1*": "/tasks-view-25/1",
+    "/assignees/teams/tasks\\?teamIds[]=22&filter=status\\:Not+started&limit=25&page=2*": "/tasks-view-25/2",
+    "/assignees/teams/tasks\\?teamIds[]=22&filter=status\\:Not+started&limit=25&page=3*": "/tasks-view-25/3",
+    "/assignees/teams/tasks\\?teamIds[]=22&filter=status\\:Not+started&limit=25&page=4*": "/tasks-view-25/4",
+    "/assignees/teams/tasks\\?teamIds[]=22&filter=status\\:Not+started&limit=25&page=5*": "/tasks-view-25/5",
+    "/assignees/teams/tasks\\?teamIds[]=22&filter=status\\:Not+started&limit=50&page=1*": "/tasks-view-50/1",
+    "/assignees/teams/tasks\\?teamIds[]=22&filter=status\\:Not+started&limit=100&page=1*": "/tasks-view-100/1",
```
Comment (devkokov) [inline json-server/config/routes.json:21]: I would use a route variable for this

`"/assignees/teams/tasks\\?teamIds[]=22&filter=status\\:Not+started&limit=25&page=:pageNumber*": "/tasks-view-25/:pageNumber"`

Comment (kate-49) [reply json-server/config/routes.json:21]: I tried exactly that but it didnt pick up the route for some reason.

Comment (devkokov) [reply json-server/config/routes.json:21]: This works for me

`"/assignees/teams/tasks\\?teamIds[]=22&filter=status\\:Not+started&limit=:limit&page=:pageNumber&*": "/tasks-view-:limit/:pageNumber",`

File: cypress/e2e/client-tasks/pagination.cy.js [hunk 3]
```diff
@@ -11,17 +11,26 @@ describe("Pagination", () => {
           .invoke('val').should('contain', 'per-page=25')
     })
 
-    it("will not show previous arrow on page 1 but will show it on other pages", () => {
+    it("will not show previous on page 1 but will show it on other pages", () => {
       cy.get(".previous-page-pagination-link").should('not.exist')
-      cy.get("#top-nav .next-page-pagination-link").click()
+      cy.get('#top-nav > :nth-child(1) > .flex-container > .govuk-pagination__list > :nth-child(2) > .govuk-link').click()
```
Comment (denialanderror) [inline cypress/e2e/client-tasks/pagination.cy.js:16]: Addressing elements this way is really brittle. If the pagination component changes its implementation or styling, this test will fail, even though all it should be interested in is behaviour.

If you can't select the element by text, the next best option is to add `data-cy` attributes.

Comment (kate-49) [reply cypress/e2e/client-tasks/pagination.cy.js]: Sorry I should've noticed this, just amended it based on the newer elements in the pagination

File: cypress/e2e/client-tasks/pagination.cy.js [hunk 4]
```diff
@@ -11,17 +11,27 @@ describe("Pagination", () => {
...
+    it("will not show previous on page 1 but will show it on other pages", () => {
       cy.get(".previous-page-pagination-link").should('not.exist')
-      cy.get("#top-nav .next-page-pagination-link").click()
+      cy.get(`[aria-label="Page 2"]`).first().click()
+      cy.get(".previous-page-pagination-link").should('be.visible', 'Previous')
+      cy.get(`[aria-label="Page 3"]`).first().click()
+      cy.get(".previous-page-pagination-link").should('be.visible', 'Previous')
+      cy.get(`[aria-label="Page 4"]`).first().click()
+      cy.get(".previous-page-pagination-link").should('be.visible', 'Previous')
+      cy.get(`[aria-label="Page 5"]`).first().click()
```
Comment (devkokov) [inline cypress/e2e/client-tasks/pagination.cy.js:22]: what's the value in checking pages 3, 4 and 5?

Comment (kate-49) [reply cypress/e2e/client-tasks/pagination.cy.js:22]: I guess checking that for some reason the previous button doesn't disappear? Happy to take it out if you think its overkill

**Review by devkokov — APPROVED** [review]

**Pull request merged**